From b1c91b9c9a24dbe4970a9d3d53e93c18aa531695 Mon Sep 17 00:00:00 2001
From: Louis Marelle <louis.marelle@latmos.ipsl.fr>
Date: Mon, 15 Oct 2018 18:32:58 +0200
Subject: [PATCH 07/71] Fix dry deposition seasonality

In chem/module_dep_simple.F, forces iseason to category 4 (winter, snow on
ground and subfreezing) if snow height >= 10 cm or if the land type is ice.
Also includes a case for the Southern Hemisphere, previously the seasonality
of dry deposition had a hard-coded asumption that the model was run in the
Northern Hemisphere for all mechanisms other than MOZART-based ones.
---
 chem/module_dep_simple.F | 36 ++++++++++++++++++++++++++----------
 1 file changed, 26 insertions(+), 10 deletions(-)

diff --git a/chem/module_dep_simple.F b/chem/module_dep_simple.F
index 8fdd765..42bab16 100755
--- a/chem/module_dep_simple.F
+++ b/chem/module_dep_simple.F
@@ -175,16 +175,6 @@ SUBROUTINE wesely_driver( id, ktau, dtstep, config_flags, current_month,  &
 
       CALL wrf_debug(15,'in dry_dep_wesely')
 
-      if( .not. chm_is_moz ) then
-         if( julday < 90 .or. julday > 270 ) then
-            iseason = 2
-            CALL wrf_debug(15,'setting iseason to 2')
-         else
-            iseason = 1
-         endif
-      end if
-
-
 tile_lat_loop : &
       do j = jts,jte 
 tile_lon_loop : &
@@ -193,7 +183,32 @@ tile_lon_loop : &
 
             iland = luse2usgs( ivgtyp(i,j) )
 !--
+            ! For mechanisms other than MOZART, set iseason based on julian day
+            if( .not. chm_is_moz ) then
+               if(xlat(i, j) >= 0) then ! Northern Hemisphere case
+                 if( julday < 90 .or. julday > 270 ) then
+                    iseason = 2
+                    CALL wrf_debug(15,'setting iseason to 2')
+                 else
+                    iseason = 1
+                 endif
+               else ! Southern Hemisphere case
+                 if( julday > 270 .or. julday < 90 ) then
+                    iseason = 2
+                    CALL wrf_debug(15,'setting iseason to 2')
+                 else
+                    iseason = 1
+                 endif
+               end if !if xlat(i, j) >= 0
+            end if
 
+            ! If snow height >= 10 cm or if the land type is ice, force iseason = 4
+            ! ("winter, snow on ground and subfreezing" category)
+            if( snowh(i,j) >= .01 .or. iland == isice_temp) then
+               iseason = 4
+            end if
+
+            ! For MOZART mechanisms, set iseason based on the external dataset seasonal_wes
             if( chm_is_moz ) then
                if( snowh(i,j) < .01 ) then 
                   iseason = seasonal_pft(id)%seasonal_wes(i,j,iland,current_month)
@@ -201,6 +216,7 @@ tile_lon_loop : &
                   iseason = 4
                endif
             end if
+
             ta    = tsk(i,j)
             rad   = gsw(i,j)
             vegfrac = vegfra(i,j)
-- 
2.31.1

